onyo/lib/onyo.py

```diff
@@ -144,11 +155,56 @@ def get_config(self,
                   KEY = VALUE
         """
 
+        value = None
+
         # repo version shim
         if self.version == '1' and key == 'onyo.assets.name-format':
             key = 'onyo.assets.filename'
 
-        return self.git.get_config(key) or self.git.get_config(key, self.onyo_config)
+        #
+        # check cache
+        #
+        cache_hit = False
+
+        try:
+            # git
+            value = self._config_cache['git'][key]
+            ui.log_debug(f"config '{key}' acquired from cache of git config: '{value}'")
+            cache_hit = True
+        except (KeyError, TypeError):
+            cache_hit = False
+            pass
+
+        if value is None:
+            # onyo
+            try:
+                value = self._config_cache['onyo'][key]
+                ui.log_debug(f"config '{key}' acquired from cache of onyo config: '{value}'")
+                cache_hit = True
+            except (KeyError, TypeError):
+                cache_hit = False
+                pass
+
+        if cache_hit:
+            return value
+
+        #
+        # query actual config files
+        #
+        ui.log_debug(f"config '{key}' cache miss")
+
+        # query the full git config stack
+        value = self.git.get_config(key)
+        self._config_cache.setdefault('git', {})[key] = value  # pyre-ignore[6]
+
+        if value is not None:
```

Comment:
Couldn't this be simply `if value:`?

Reply:
No. What if `value` is `''` or `0`?

Reply:
@TobiasKadelka : This would break the moment `self.git.get_config()` returns python types rather than just strings. A config may be set  to `"false"` and the getter could return `False`, since this is in fact what we'd want to deal with. So, your condition would trigger b/c it can't distinguish between nothing being returned and `False` being returned.